README.md

```diff
@@ -130,6 +130,10 @@ Setting `disableFeatures: ['random', 'stdio', 'clocks']` will disable all featur
 
 Note that pure components **will not report errors and will instead trap**, so that this should only be enabled after very careful testing.
 
+The features that are not included by default are:
+* `'http'` - Support for sending and receiving HTTP requests, depends on `wasi:io` 
+* `'fetch-event'` - Enables using `fetchEvent` to respond to `wasi:http/incoming-handler@0.2.0#handle`. If the target world does note export `wasi:http/incoming-handler@0.2.0`, this will be ignored.
```

Comment:
Just like we automatically merge WASI imports into the target world, perhaps we can automatically merge the `wasi:http/incoming-handler@0.2.0` export into the target world when this feature is used?

Reply:
Actually isn't this the default behaviour anyway? Or do we still strip the export if it's not explicitly in the target world?

Reply:
With this PR we want to only optionally strip the export from the StarlingMonkey engine [here](https://github.com/bytecodealliance/ComponentizeJS/blob/544b4edd28cb49d91d88b1abcd333712c65ecf91/crates/spidermonkey-embedding-splicer/src/splice.rs#L67-L74) based on the value of the feature. My intention with that statement is to make it such that, if the target world does not export the interface but the feature is enabled, we should probably still strip the export.

Reply:
Shouldn't it read "If the target world does export" though? Rather than "does not". Thinking about this further we probably want to make this an explicit error. That is, if you enable this feature and the target world already exports the incoming handler, then we should throw an error that you should either not target this export, or you should use the fetch event version of it, but not both.

Reply:
I think there is a disconnect in what we each mean by the target world. I am talking about the `wit` definition, not the guest content. As in, consider the following world

```
package local:hello;

world hello {
  export hello: func(name: string) -> string;
}
```
and the `fetch-event` is set, if we do not strip the export from the engine, it will end up in the output component but maybe that i fine?  

I also think that there is currently no way to check if a handler is attached to the fetch event but we can check if there is something that targets the incoming handler export so erroring on that would be good, I agree.

Reply:
We are referring to the same thing. I'm saying that in the above, we do merge in the WASI world automatically even if it wasn't originally targeted. But that if it is explicitly targeted, the user might want to write their own custom handler something like the test case in https://github.com/bytecodealliance/jco/blob/main/test/fixtures/componentize/wasi-http-proxy/source.js#L102.

Reply:
I had written a long comment explaining why it felt too magical to me to have this option automatically add the `incoming-handler` export, but in writing it up I've come around and now think this is fine to do :)

The key insight is that we're talking about an export, not an import: it's valid to export interfaces/functions that aren't part of the target world. It means that those exports won't be called, but that's fine.

Reply:
In case of writing an app that will be  used as a http trigger using the fetch event, wouldn't the host expect to call the export even though it is not part of the target world explicitly?

Reply:
If the developer is specifying a target world that is different from what their use case actually needs, and their runtime provides, then that's a bug in the configuration. With what Guy is proposing here, there'd be a higher likelihood of things sort of accidentally working out regardless. That could be seen as a downside, but I'm not even entirely sure it is, honestly.

Reply:
Maybe I am misunderstanding here. To verify my understanding - the proposal above is to not retain `fetchEvent` incase the target world contains the `wasi:http/incoming-handler` export and only have the feature take effect if it is not explicitly part of the target world.

Reply:
Oh, I see—I had missed that part of it. On that, I agree with you: I think this setting should be usable whether the targeted world includes `incoming-handler` or not.

@guybedford doing what you proposed would make it substantially harder to use `FetchEvent`, because now anyone who wants to target an environment that supports `wasi:http/proxy` has to edit the target world not to include `incoming-handler` if they want to use `FetchEvent`. I think that's a requirement we really shouldn't put on people—and it'd be quite hard for us to satisfy using Spin as a development tool.

Reply:
I am happy to update this PR once there is consensus on this discussion.

Reply:
Guy and I just discussed this further, and came to the realization that there's a cleaner way to do this: with #117, ComponentizeJS will only generate exports for symbols exported at the JS level, too. Based on that, we could simply say "if after treeshaking we still would export the symbol, then we'll do so. Otherwise, we'll leave it alone."

In effect, that'd mean that if the input StarlingMonkey already includes an export, and content doesn't specify that same export, ComponentizeJS will leave the original export alone.

@karthik2804, if that makes sense to you, then changing #117 accordingly should make it so that this issue isn't needed anymore.

Reply:
I think I am roughly tracking it. Currently, #117 is purely for imports. I will take a stab at updating it and see if I understand things correctly and report back.